modules/core/src/main/java/org/apache/ignite/internal/processors/job/GridJobProcessor.java

```diff
@@ -2020,6 +2020,14 @@ else if (removeFromActive(jobWorker)) {
                         rwLock.readUnlock();
                     }
                 }
+
+                if (ctx.performanceStatistics().enabled()) {
```

Comment:
Why we do this inside `if (!worker.isInternal()) {`?

Reply:
Fixed.